### DISCUSSION TITLE
ERROR:  42883 - Querying individual row to HTTP Request and returning the results inside of it's own column

### DISCUSSION THREAD

**Discussion by bosoleil** [supabase/supabase#26860] [Questions] [answered]
Hey guys,

I have a column called email and it's a text column that is also the primary key column. I already have a verify_email function that works when I test it in the SQL editor. Now I'm trying to have this function perform in each and individual emails (44k) of an entire table and have the results returned in another column. I keep having issues with it and I'm not sure how to fix that. 

My Error (Removed my apikey)

[](ERROR:  42883: function http(unknown, text, unknown) does not exist
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
QUERY:  SELECT *                       FROM http(
    'GET',
    'https://api.emailable.com/v1/verify?email=' || NEW.email || '&api_key=live_YOURAPIKEY',
    '{}'
  )
CONTEXT:  PL/pgSQL function validate_email() line 4 at SQL statement)

```
create extension if not exists pgcrypto;
create extension if not exists http;

-- Create or replace the function to verify emails
create
or replace function verify_email (email_address text) returns jsonb language plpgsql as $$
DECLARE
    api_key text := 'YOUR API KEY';
    api_url text := 'https://api.emailable.com/v1/verify?email=' || email_address || '&api_key=' || api_key;
    response text;
    response_json jsonb;
BEGIN
    -- Make the HTTP GET request
    SELECT content INTO response
    FROM http_get(api_url);

    -- Try to parse the response into JSON
    BEGIN
        response_json := response::jsonb;
    EXCEPTION WHEN others THEN
        -- In case of an error, return a JSON object indicating the failure
        RETURN jsonb_build_object('error', 'Failed to parse JSON response', 'response', response);
    END;

    RETURN response_json;
END;
$$;

-- Add a new column to store the verification results if it doesn't exist
do $$
BEGIN
    IF NOT EXISTS (SELECT 1 FROM information_schema.columns 
                   WHERE table_name='merged_asana_leads' and column_name='verification_result') THEN
        ALTER TABLE merged_asana_leads ADD COLUMN verification_result jsonb;
    END IF;
END
$$;

-- Update the new column with the email verification results
update merged_asana_leads
set
  verification_result = verify_email (email);
```

**Comment by GaryAustin1** [comment]
You show a verify_email function.
But your error appears to be on a validate_email function that is a trigger (as you use NEW in it).
![image](https://github.com/supabase/supabase/assets/54564956/e086badf-beca-483d-876c-edc0b0f6dab9)
The error is you are calling http incorrectly in it.

**Reply by bosoleil** [reply]
Hey @GaryAustin1 thanks for the quick reply - I'm not sure to know how i should fix this? Sorry? I'm not sure. 

I'm trying to take emails from the columns 'email' validate it and return the result in the new column 'verification_result' in every rows of the email column.

**Reply by GaryAustin1** [reply]
I'm not looking at what you are trying to do.
Your error seems to be coming from a function validate_email which you don't show.

**Reply by bosoleil** [reply]
CREATE OR REPLACE FUNCTION verify_email(email_address text) 
RETURNS jsonb LANGUAGE plpgsql AS $$
DECLARE
    api_key text := 'APIKEY';
    api_url text := 'https://api.emailable.com/v1/verify?email=' || email_address || '&api_key=' || api_key;
    response jsonb;
BEGIN
    SELECT http_get(api_url) INTO response;
    RETURN response;
END;
$$;
-- Example of how to call the function
SELECT verify_email('example@example.com');

**Reply by bosoleil** [reply]
This is my function  - thank you so much for taking the time btw

**Reply by GaryAustin1** [reply]
That is not the function getting the error.  A function validate_email is getting the error not verify_email.  It is like you have a trigger function you have forgotten about.
![image](https://github.com/supabase/supabase/assets/54564956/eca66b1b-c864-458f-a2f0-4957e0082b3c)

**Reply by GaryAustin1** [reply, marked as answer]
Solved here: https://github.com/orgs/supabase/discussions/26871#discussioncomment-9599054
There was another function incorrectly calling http extension.
